src/replication.c

```diff
@@ -1324,20 +1382,43 @@ check_for_command_completion(spec_t *spec, rcommon_cmd_t *rcomm_cmd, ISTGT_LU_CM
 		}
 	}
 
+	response_received = success + failure;
+	min_response = MAX(rcomm_cmd->replication_factor -
+	    rcomm_cmd->consistency_factor + 1, rcomm_cmd->consistency_factor);
+
 	if (rcomm_cmd->opcode == ZVOL_OPCODE_READ) {
-		if ((rcomm_cmd->copies_sent != (success + failure))) {
-			rc = 0;
-		} else if (rcomm_cmd->copies_sent == failure) {
-			rc = -1;
+		response_received = success + failure;
+
+		if (rcomm_cmd->copies_sent >= min_response) {
+			if (response_received < min_response) {
```

Comment:
in this case i.e., copies_sent >= min_response, just checking for success condition and failure condition should be good enough. So:
- does success received >= min_response, if so, set rc as 1 and do handle_read_consistency
- else if response_received == copies_sent, set rc as -1 (becz required number of success is not received)
- anything else, would be rc as 0, which means, we need to wait for more responses

Reply:
ok